tom_tns/templatetags/tns_extras.py

```diff
@@ -34,24 +37,38 @@ def report_to_tns(context):
         initial['reporter'] = reporter
 
     reduced_datum = None
-    if 'datum' in context:
+    if 'datum' in context and context['datum'].data_type == 'photometry':
         reduced_datum = context['datum']
     else:
         # Get photometry if available
         photometry = target.reduceddatum_set.filter(data_type='photometry')
         if photometry.exists():
             reduced_datum = photometry.latest()
     if reduced_datum:
-        initial['observation_date'] = reduced_datum.timestamp
-        initial['flux'] = reduced_datum.value.get('magnitude')
-        initial['flux_error'] = reduced_datum.value.get('error')
-        filter_name = reduced_datum.value.get('filter')
-        mapped_filter = map_filter_to_tns(filter_name)
-        if mapped_filter in TNS_FILTER_IDS:
-            initial['filter'] = (TNS_FILTER_IDS[mapped_filter], mapped_filter)
-        instrument_name = reduced_datum.value.get('instrument')
-        if instrument_name in TNS_INSTRUMENT_IDS:
+        hermes_datum_converter = get_hermes_data_converter_class()(validate=False)
+        phot_data = hermes_datum_converter.get_hermes_photometry(reduced_datum)
+        initial['observation_date'] = phot_data['date_obs']
+        if phot_data.get('exposure_time'):
+            initial['exposure_time'] = phot_data['exposure_time']
+        instrument_name = map_instrument_to_tns(phot_data.get('instrument', ''))
```

Comment:
This should try the "telescope" field in the data as well if the instrument field doesn't work.
Most astronomers will be able to map LCO 1m to 'LCO1m - Sinistro' etc. for the telescopes they use regularly.

Reply:
The instrument field is a requirement for TNS report submissions... Are you suggesting we try to use the value of a "telescope" field if there is no instrument field found in the reduced datum? I think that is kind of the domain of the HermesDataConverter, where in a custom implementation you could map your datums telescope key to the "instrument" key in the hermes schema and it would be used here.

Reply:
I was suggesting that the `map_instrument_to_tns` could check the `telescope` field as well if it didn't find any matches in `instrument` so I could use that setting for either. For many observers these fields are redundant in the Reduced Datum, and it would be reasonable to translate either into a tns instrument field.
This doesn't mean that it should be used for the HERMES instrument field for the Hermes message.